subsys/bluetooth/host/hci_core.c

```diff
@@ -3798,9 +3798,9 @@ int bt_disable(void)
 
 	bt_monitor_send(BT_MONITOR_CLOSE_INDEX, NULL, 0);
 
-#if defined(CONFIG_BT_EXT_ADV) || defined(CONFIG_BT_BROADCASTER)
+#if defined(CONFIG_BT_BROADCASTER)
 	bt_adv_reset_adv_pool();
-#endif /* CONFIG_BT_EXT_ADV || CONFIG_BT_BROADCASTER */
+#endif /* CONFIG_BT_BROADCASTER */
```

Comment:
Since this part is fixing something in the host and not a test case, wouldn't it make sense to split it into a separate commit?

Reply:
In hindsight probably yes, the bug got caught in a unittest though. If a 2nd request for splitting it out comes in I'll put it in a separate commit, otherwise I'll leave it as is

Reply:
I would agree with Johan that it's best to split it. 

I won't block this PR based on that though

Reply:
I'll split